lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -152,11 +152,13 @@ local scroll_fn = function(self, direction)
 end
 
 previewers.file_maker = function(filepath, bufnr, opts)
-  opts = opts or {}
-  opts.preview = opts.preview or {}
+  opts = vim.F.if_nil(opts, {})
+  -- TODO(conni2461): here shouldn't be any hardcoded magic numbers ...
+  opts.preview = vim.F.if_nil(opts.preview, {})
```

Comment:
wouldn't this still set `opts.preview` to `true` or `false` when it's non nil, and then the following dictionary lookups would still fail?

Reply:
no because the data is already changed at this point in time, we do that on previewer instanzing 

https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/previewers/buffer_previewer.lua#L38-L46

Do you still experience #2101 with this PR?